src/main/java/com/openshift/restclient/capability/resources/IPropertyAccessCapability.java

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package com.openshift.restclient.capability.resources;
+
+import java.util.Map;
+
+import com.openshift.restclient.capability.ICapability;
+
+/**
+ * A mechanism to access the underlying content of a json
+ * structure
+ * 
+ * Relies on a dot delimited path to indentify the root of the entity
+ * being returned (e.g. 'metadata.labels').  This does not provide
+ * a mechanism to retrieve properties across an array kind.
```

Comment:
I think I agree with @adietish that we want this API as minimal as possible.  It's a last resort to access something that isn't exposed with a proper getter.